src/web3-utils.js

```diff
@@ -125,6 +128,54 @@ export async function getIsContractAccount(web3, account) {
   }
 }
 
+const gasPriceApi = 'https://ethgasstation.info/json/ethgasAPI.json'
+export async function getGasPrice({
+  mainnet: { safeMinimum = '3', disableEstimate } = {},
+} = {}) {
+  if (network.type !== 'main') {
+    // Hardcode 10 for non-mainnet networks
+    return toWei('10', 'gwei')
+  }
+
+  // Mainnet and gas estimation not disabled
+  if (!disableEstimate) {
+    const safeMinimumInWei = toWei(safeMinimum, 'gwei')
+    let priceInWei
+
+    try {
+      const response = await fetch(gasPriceApi, {
+        method: 'GET',
+        mode: 'cors',
+        cache: 'no-cache',
+      })
+      const jsonResponse = await response.json()
+      // Note that all prices from ethgasstation need to be divided by 10 to be in gwei.
+      // The response contains a list of suggested gas prices from 2-120 in
+      // gasPriceRange, so 40 is "slightly higher than the recommended price".
+      const fasterPrice = (jsonResponse.gasPriceRange[40] || 0) / 10
+      // Just in case, if this isn't available or is way too high,
+      // prefer the suggested safe low price.
+      const safePrice = (jsonResponse.safeLow || 0) / 10
+      const recommendedPrice =
+        fasterPrice < safePrice || fasterPrice - safePrice > 10
```

Comment:
When `fasterPrice - safePrice > 10`, should we use `safePrice + 10` rather than `safePrice`?

Reply:
I'd just stick with the `safePrice`; if it's available, it should mine in a reasonable amount of time.

Reply:
mmm I find it a bit odd that we would accept `safePrice + 10`, but revert to `safePrice` as soon as we reach `safePrice + 11`. Shouldn't we just use `safePrice` then?

Reply:
Hmm yes, I see your point. Let's change it to be `safePrice + 10`

Reply:
Then we could write it this way :smile: 

```jsx
const recommendedPrice = Math.max(safePrice, Math.min(fasterPrice, safePrice + 10))
```

I’m not sure why `fasterPrice` would be lower than `safePrice` though, is it just to be safe because it comes from an external API?

Reply:
> I’m not sure why fasterPrice would be lower than safePrice though, is it just to be safe because it comes from an external API?

Yep, just in case something really weird happens...

Reply:
Fixed in 404c31e42eb2b04d249f17879aa0f79dfbe80b1c